docs/start/getting-started/fragments/flutter/integrate.md

```diff
@@ -1,120 +1,470 @@
+
+### Querying for Todos
+
+Now that we have a way to add Todo items, we need a way to list them out.
+
+Update the `_initializeApp()` function in the `_TodosPageState` class.
+
+```dart
+Future<void> _initializeApp() async {
+
+  // configure Amplify
+  await _configureAmplify();
+
+  // fetch Todo entries from DataStore
+  await _fetchTodos();
+
+  // after both configuring Amplify and fetching Todo entries, update loading
+  // ui state to loaded state
+  setState(() {
+    _isLoading = false;
+  });
+}
+```
+
+Update the `_fetchTodos()` function in the `_TodosPageState` class. For the purposes of this tutorial, we will be querying for all stored **Todo** entries, but you can perform more advanced filtering, sorting or even pagination as well. You can [learn more](~/lib/datastore/data-access.md/q/platform/flutter#query-data) about DataStore data queries.
+
+```dart
+Future<void> _fetchTodos() async {
   try {
+  
+    // query for all Todo entries by passing the Todo classType to
+    // Amplify.DataStore.query()
+    List<Todo> updatedTodos = await Amplify.DataStore.query(Todo.classType);
+    
+    // update the ui state to reflect fetched todos
     setState(() {
-      _amplifyConfigured = true;
+      _todos = updatedTodos;
     });
   } catch (e) {
-    print(e);
+    print('An error occurred while querying Todos: $e');
   }
+}
+```
 
+If you restart your app now, you should see your added todos start to show up on the list! However, if you add more entries, they don’t show up until you refresh the app. Let’s wire up the real-time data observation next.
+
+### Observing updates to Todos
+
+Amplify DataStore provides a way to subscribe to data updates to your model. We will use this to listen for change events and update our list accordingly.
+
+To the top of the `_TodosPageState` class, add a `StreamSubscription`.
+
+```dart
+class _TodosPageState extends State<TodosPage> {
+
+  // subscription to Todo model update events
+  StreamSubscription _subscription;
+```
+
+Update the `dispose()` function in the `_TodosPageState` class.
+
+```dart
+@override
+void dispose() {
+
+  // cancel the subscription when the state is removed from the tree
+  _subscription.cancel();
+  super.dispose();
+}
+```
+
+Update the `_fetchTodos()` function in the `_TodosPageState` class.
+
+```dart
+Future<void> _initializeApp() async {
+
+  // configure Amplify
+  await _configureAmplify();
+  
+  // listen for updates to Todo entries by passing the Todo classType to
+  // Amplify.DataStore.observe() and when an update event occurs, fetch the
+  // todo list
+  // 
+  // note this strategy may not scale well with larger number of entries 
+  _subscription = Amplify.DataStore.observe(Todo.classType).listen((event) {
+    _fetchTodos();
+  });
+
+  // fetch Todo entries from DataStore
+  await _fetchTodos();
```

Comment:
I think we should avoid this call pattern in our getting started guide. This is not really the right intended usage of `observe()` API and if customers are not careful and leave this in, it will be pretty bad for performance. Also what's the reason for calling `_fetchTodos()` twice here?

Reply:
I tried to call out that this isn't optimal for larger datasets but, for the purposes of getting a simple app up and running, I think it's the simplest way to get the app displaying real time updates without getting too deep into the weeds about how to set up their app to actually respond to the granular events. I agree it would be bad for customers to leave things as is but I think the purpose of the guide isn't necessarily to help them with more advanced scenarios or UX patterns but to convey the value of Amplify quickly. I believe *everything* in this tutorial could/should be shored up as customers expand upon their use cases.

> Also what's the reason for calling _fetchTodos() twice here?

The `_fetchTodos()` on line 409 only fires in response to future streamed events while the one on on line 413 is part of the cold start.

Reply:
>I agree it would be bad for customers to leave things as is but I think the purpose of the guide isn't necessarily to help them with more advanced scenarios or UX patterns but to convey the value of Amplify quickly.

We should assume the worst in my opinion though it's not a security risk but a performance one, but still significant. Observe is not necessarily an advanced case and if it is we should just leave it out. For local updates, we can update the UI for every `save` and call it quits. If we do want to show cross device sync, we should demonstrate `observe` accurately.

Reply:
Hmm given that I think the value of cross device sync is quite large with DataStore, it's probably worth leaving in, imo. I'm not sure that just updating UI for every `save` is enough to show that things are actually being persisted since we'd essentially just be optimistically updating the UI that way. Could you please share what you mean with demonstrating usage of `observe` accurately?